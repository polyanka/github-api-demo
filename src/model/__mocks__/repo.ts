import { IRepo } from '../repo';
export const repos: IRepo[] = [
  {
    id: 1,
    name: 'name1',
    description: 'description1',
    created_at: '2021-08-30T21:19:21Z',
    html_url: 'https://www.example1.com',
    stargazers_count: 5,
    watchers_count: 3,
    language: 'JavaScript',
  },
  {
    id: 2,
    name: 'name2',
    description: 'description2',
    created_at: '2021-09-30T22:56:01Z',
    html_url: 'https://www.example2.com',
    stargazers_count: 1,
    watchers_count: 12,
    language: 'TypeScript',
  },
];
