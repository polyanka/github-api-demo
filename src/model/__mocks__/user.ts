import { IUser } from '../user';

export const user: IUser = {
  avatar_url: 'https://www.example.com/avatar_url.png',
  id: 1,
  login: 'Login',
  name: 'User',
  blog: 'https://www.example-blog.com',
  location: null,
  public_repos: 0,
  public_gists: 2,
  followers: 2,
  following: 0,
  company: null,
};
