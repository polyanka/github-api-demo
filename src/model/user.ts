export interface IUserBase {
  id: number;
  login: string;
  avatar_url: string;
}

export interface IUsers {
  users: IUserBase[];
}

export interface IUser extends IUserBase {
  name: string;
  blog: string;
  location: string;
  public_repos: number;
  public_gists: number;
  followers: number;
  following: number;
  company: string;
}
