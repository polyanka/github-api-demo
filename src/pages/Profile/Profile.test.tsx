import { user } from '@src/model/__mocks__/user';
import { shallow } from 'enzyme';
import * as React from 'react';
import * as ReactRedux from 'react-redux';
import ReactRouter from 'react-router';

import { ProfilePage } from './index';

const login = 'polyanka';

describe('<ProfilePage />', () => {
  beforeEach(() => {
    jest.spyOn(ReactRouter, 'useParams').mockImplementation(() => ({
      login,
    }));
    jest.spyOn(ReactRedux, 'useDispatch').mockReturnValue(jest.fn());
  });

  it('start render <ProfilePage />', () => {
    jest.spyOn(ReactRedux, 'useSelector').mockReturnValue({
      user: null,
      loading: true,
      error: false,
    });

    const wrapper = shallow(<ProfilePage />);

    expect(wrapper.find('Spiner')).toHaveLength(1);
    expect(wrapper.find('ErrorMessage')).toHaveLength(0);
    expect(wrapper.find('NotFound')).toHaveLength(0);
    expect(wrapper.find('UserDataList')).toHaveLength(0);
  });

  it('renders correctly ProfilePage component with user data', () => {
    jest.spyOn(ReactRedux, 'useSelector').mockReturnValue({
      user,
      loading: false,
      error: false,
    });

    const wrapper = shallow(<ProfilePage />);

    expect(wrapper.find('Spiner')).toHaveLength(0);
    expect(wrapper.find('ErrorMessage')).toHaveLength(0);
    expect(wrapper.find('NotFound')).toHaveLength(0);
    expect(wrapper.find('UserDataList')).toHaveLength(1);
    expect(wrapper.find('ReposList')).toHaveLength(1);

    expect(wrapper).toMatchSnapshot();
  });

  it('render ErrorMessage', () => {
    jest.spyOn(ReactRedux, 'useSelector').mockReturnValue({
      user: null,
      loading: false,
      error: true,
    });

    const wrapper = shallow(<ProfilePage />);

    expect(wrapper.find('Spiner')).toHaveLength(0);
    expect(wrapper.find('ErrorMessage')).toHaveLength(1);
    expect(wrapper.find('NotFound')).toHaveLength(0);
    expect(wrapper.find('UserDataList')).toHaveLength(0);
    expect(wrapper.find('ReposList')).toHaveLength(0);
  });

  it('render NotFound', () => {
    jest.spyOn(ReactRedux, 'useSelector').mockReturnValue({
      user: null,
      loading: false,
      error: false,
    });

    const wrapper = shallow(<ProfilePage />);

    expect(wrapper.find('Spiner')).toHaveLength(0);
    expect(wrapper.find('ErrorMessage')).toHaveLength(0);
    expect(wrapper.find('NotFound')).toHaveLength(1);
    expect(wrapper.find('UserDataList')).toHaveLength(0);
    expect(wrapper.find('ReposList')).toHaveLength(0);
  });
});
