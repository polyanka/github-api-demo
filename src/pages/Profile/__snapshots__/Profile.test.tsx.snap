// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProfilePage /> renders correctly ProfilePage component with user data 1`] = `
<WithStyles(ForwardRef(Grid))
  container={true}
  spacing={4}
>
  <WithStyles(ForwardRef(Grid))
    item={true}
    md={4}
    xs={12}
  >
    <UserDataList
      user={
        Object {
          "avatar_url": "https://www.example.com/avatar_url.png",
          "blog": "https://www.example-blog.com",
          "company": null,
          "followers": 2,
          "following": 0,
          "id": 1,
          "location": null,
          "login": "Login",
          "name": "User",
          "public_gists": 2,
          "public_repos": 0,
        }
      }
    />
  </WithStyles(ForwardRef(Grid))>
  <WithStyles(ForwardRef(Grid))
    alignItems="stretch"
    container={true}
    direction="column"
    item={true}
    md={8}
    xs={12}
  >
    <WithStyles(ForwardRef(Typography))
      gutterBottom={true}
      variant="h5"
    >
      Repositories
    </WithStyles(ForwardRef(Typography))>
    <ReposList
      user={
        Object {
          "avatar_url": "https://www.example.com/avatar_url.png",
          "blog": "https://www.example-blog.com",
          "company": null,
          "followers": 2,
          "following": 0,
          "id": 1,
          "location": null,
          "login": "Login",
          "name": "User",
          "public_gists": 2,
          "public_repos": 0,
        }
      }
    />
  </WithStyles(ForwardRef(Grid))>
</WithStyles(ForwardRef(Grid))>
`;
